OpenEdXMobile/src/main/java/org/edx/mobile/base/MainApplication.java

```diff
@@ -132,16 +130,32 @@ private void init() {
         );
     }
 
-    private boolean needVersionUpgrade(Context context) {
-        boolean needVersionUpgrade = false;
-        PrefManager.AppInfoPrefManager pmanager = new PrefManager.AppInfoPrefManager(context);
-        Long previousVersion = pmanager.getAppVersionCode();
-        final int curVersion = BuildConfig.VERSION_CODE;
-        if (previousVersion < curVersion) {
-            needVersionUpgrade = true;
-            pmanager.setAppVersionCode(curVersion);
+    private void checkIfAppVersionUpgraded(Context context) {
+        PrefManager.AppInfoPrefManager prefManager = new PrefManager.AppInfoPrefManager(context);
+        long previousVersionCode = prefManager.getAppVersionCode();
+        final long curVersionCode = (long) BuildConfig.VERSION_CODE;
+        if (previousVersionCode < 0) {
```

Comment:
If the app is opened for the first time, shouldn't there be a deterministic value for `previousVersionCode`?